libs/sdk-ui-kit/src/Dialog/StylingEditorDialog/StylingEditorDialog.tsx

```diff
@@ -64,32 +63,66 @@ const StylingEditorDialogCore = <T extends StylingPickerItemContent>(props: ISty
         disableSubmit,
         showProgressIndicator,
     } = props;
+    const intl = useIntl();
     const providedExamples = !!examples && examples.length !== 0 && !!exampleToColorPreview;
-    const [nameField, setNameField] = useState(stylingItem?.name ?? "");
 
-    const [definitionField, setDefinitionField] = useState(
-        stylingItem?.content ? JSON.stringify(stylingItem.content, null, 4) : "",
-    );
-    const [invalidDefinition, setInvalidDefinition] = useState(false);
-    const isSubmitDisabled = useMemo(
-        () => nameField === "" || definitionField === "" || invalidDefinition || disableSubmit,
-        [nameField, definitionField, invalidDefinition, disableSubmit],
-    );
-    const intl = useIntl();
-    const validateDefinition = (value: string): boolean => {
+    const initialNameField = stylingItem?.name ?? "";
+    const [nameField, setNameField] = useState({
+        value: initialNameField,
+        changed: false,
+    });
+
+    const handleNameChange = (value: string) => {
+        setNameField({
+            changed: value !== initialNameField,
+            value: value,
+        });
+    };
+
+    const initialDefinitionField = stylingItem?.content ? JSON.stringify(stylingItem?.content, null, 4) : "";
+    const [definitionField, setDefinitionField] = useState({
+        value: initialDefinitionField,
+        changed: false,
+        valid: true,
+    });
+
+    const handleDefinitionChange = (value: string) => {
+        const newState = {
+            value: value,
+            changed: false,
+            valid: true,
+        };
```

Comment:
I find the state and logic quite complicated at first glance. I am thinking whether just setting one state variable like `errorMessage` would not be a simpler approach. The message would always render the variable's content and you would also know when to disable the save button. You could then get rid of the `valid` keys in both states. Looks like the `change` key does not to be in state if I'm not mistaken. Not sure if this would cover everything, but it's just an idea.